frontend/src/components/reflection/subjectArea/subjectAreaUtils.ts

```diff
@@ -0,0 +1,43 @@
+import { checkmarkCircle, checkmarkCircleOutline } from 'ionicons/icons'
+
+import { SubTopicItem } from './subjectArea'
+
+export function totalDone(subListItems: SubTopicItem[]): number {
+  return subListItems.filter(({ done }) => done).length
+}
+
+export function badgeStatus(
+  subListItems: SubTopicItem[],
+): 'medium' | 'primary' | 'success' {
+  const done = totalDone(subListItems)
+  const total = subListItems.length
+  return done === 0 ? 'medium' : done !== total ? 'primary' : 'success'
+}
+
+export function currentSelectedTopic(
+  currentSelected: string | undefined,
+  id: string,
+): 'tertiary' | undefined {
+  return currentSelected === id ? 'tertiary' : undefined
+}
+
+export function textLable(
+  currentSelected: string | undefined,
+  id: string,
+): 'medium' | undefined {
+  return currentSelected === undefined || currentSelected === id
+    ? undefined
```

Comment:
```suggestion
    ? null
```
Same here

Reply:
i cant use null, because the components want either a value or undefined, while null !== undefined i get type-errors.

`
(property) IonIconProps.color?: string | undefined
Type 'null' is not assignable to type 'string | undefined'.ts(2322)
`